vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -455,7 +463,14 @@ def __init__(
             self.tp_rank = 0
             self.ep_size = self.tp_size * self.dp_size
             self.tp_size = 1
-
+            if (envs.VLLM_SHARED_EXPERT_FUSION_REPLICAS > 0 and self.ep_size
+                    != envs.VLLM_SHARED_EXPERT_FUSION_REPLICAS):
+                logger.warning(
+                    "With EP enabled and share expert fusion enabled"
+                    ", share expert replica should be same as ep_size"
+                    "got share expert replica = %d"
+                    "and ep_size = %d",
+                    envs.VLLM_SHARED_EXPERT_FUSION_REPLICAS, ep_size)
```

Comment:
A few things here:

1. shouldn't this be 
```
if (envs.VLLM_SHARED_EXPERT_FUSION_REPLICAS > 0 and self.ep_size
                    != envs.VLLM_SHARED_EXPERT_FUSION_REPLICAS + 1):
```

2. This raises a `TypeError`: `TypeError: %d format: a real number is required, not NoneType`

3. I think EP will be the main way to run MoE models going forward. Especially because of this warning and to generally improve the UX, I think we should change `VLLM_SHARED_EXPERT_FUSION_REPLICAS` to be a bool where vLLM selects the number of replicas, rather than requiring it be a number selected by the user.